### DISCUSSION TITLE
Change in general IPM vignette

### DISCUSSION THREAD

**Discussion by Aariq** [levisc8/ipmr#49] [Q&A] [answered]
I noticed you recently fixed a mistake in the General IPM vignette (b6af2326249b856a51628fb944b9797c48c04d78) and I changed the General IPM I'm working on accordingly.  To my surprise, the output of that IPM _changed_.  I'm just curious what it was doing before when it was trying to integrate a discrete class over a state variable it didn't have.  I wasn't expecting that fix to have any effect.  It might be a good idea to add a check if a kernel is discrete (`family = "D_"`)  and throw an error if it has a `d_<state var>` in the formula.

btw, that vignette is soooooo helpful 🙏🏼.

**Comment by levisc8** [comment, marked as answer]
There should be a change, as the integration with midpoint rule just multiplies the expression by the value of the bin width (i.e. `d_z`). In this case, I think it's a pretty small number. 

As for the broader checking, what I really need to do is just remove the need for users to supply `d_z`s to begin with. Early on, I didn't fully understand how the integration was functioning in the model (i.e. always integrating w/r/t to initial states). In short, if you're starting on "D", then no integration, and if you're starting on "C", then it should be `d_<state_start>`. It's a fairly straightforward thing to implement, I just haven't gotten around to doing it yet. 

I'd probably keep a check for `d_z` for backward compatibility - it'd suck if this update for convenience broke a model you'd already made! Hopefully will have time for this in the not-so-distant future!

PS Sorry that the discussion tab hasn't taken off!

**Reply by Aariq** [reply]
Thanks for the quick reply!
